src/main/java/frc/robot/subsystems/elevator/factories/RealElevatorConstants.java

```diff
@@ -0,0 +1,97 @@
+import frc.robot.hardware.motor.sparkmax.SparkMaxWrapper;
+import frc.robot.hardware.request.cansparkmax.SparkMaxAngleRequest;
+import frc.robot.hardware.signal.supplied.SuppliedAngleSignal;
+import frc.robot.hardware.signal.supplied.SuppliedDoubleSignal;
+import frc.robot.subsystems.elevator.ElevatorConstants;
+import frc.robot.subsystems.elevator.ElevatorMotorStuff;
+import frc.robot.subsystems.elevator.ElevatorStuff;
+import frc.utils.AngleUnit;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+public class RealElevatorConstants {
+
+	private final static double DEBOUNCE_TIME_SECONDS = 0.05;
+
+	private final static Debouncer.DebounceType DEBOUNCE_TYPE = Debouncer.DebounceType.kBoth;
+
+	private final static CANSparkBase.SoftLimitDirection SOFT_LIMIT_DIRECTION = CANSparkBase.SoftLimitDirection.kReverse;
+
+	private final static Rotation2d REVERSE_SOFT_LIMIT_VALUE = Rotation2d.fromRotations(0);
+
+	private final static double ROTATIONS_TO_METERS_CONVERTION_RATIO = 1;
+
+	private final static SparkLimitSwitch.Type REVERSE_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyOpen;
+
+	public static ElevatorFeedforward FEEDFORWARD_CALCULATOR = new ElevatorFeedforward(0, 0, 0, 0);
+
+	public static double feedforwardCalculation(Rotation2d velocity) {
+		return RealElevatorConstants.FEEDFORWARD_CALCULATOR.calculate(velocity.getRadians());
+	}
+
+	public static void configureMotors(SparkMaxWrapper mainSparkMaxWrapper, SparkMaxWrapper secondarySparkMaxWrapper, String logPath) {
+		mainSparkMaxWrapper.setSoftLimit(SOFT_LIMIT_DIRECTION, (float) REVERSE_SOFT_LIMIT_VALUE.getRotations());
+		mainSparkMaxWrapper.enableSoftLimit(SOFT_LIMIT_DIRECTION, true);
+		mainSparkMaxWrapper.getEncoder().setPositionConversionFactor(ElevatorConstants.GEAR_RATIO);
+		mainSparkMaxWrapper.getEncoder().setVelocityConversionFactor(ElevatorConstants.GEAR_RATIO);
+		mainSparkMaxWrapper.getPIDController().setP(1);
+		mainSparkMaxWrapper.getPIDController().setI(0);
+		mainSparkMaxWrapper.getPIDController().setD(0);
+
+		secondarySparkMaxWrapper.setSoftLimit(SOFT_LIMIT_DIRECTION, (float) REVERSE_SOFT_LIMIT_VALUE.getRotations());
+		secondarySparkMaxWrapper.enableSoftLimit(SOFT_LIMIT_DIRECTION, true);
+	}
+
+	public static Pair<ElevatorMotorStuff, SparkMaxWrapper> generateMotor(String logPath, String motorName, SparkMaxDeviceID motorID) {
+		SparkMaxWrapper sparkMaxWrapper = new SparkMaxWrapper(motorID);
+		ControllableMotor motor = new BrushlessSparkMAXMotor(logPath, sparkMaxWrapper, new SysIdRoutine.Config());
+
+		Supplier<Double> motorPosition = () -> sparkMaxWrapper.getEncoder().getPosition();
+		SuppliedAngleSignal motorPositionSignal = new SuppliedAngleSignal(motorName + " angle", motorPosition, AngleUnit.ROTATIONS);
+
+		Supplier<Double> motorsVoltage = sparkMaxWrapper::getVoltage;
+		SuppliedDoubleSignal motorVoltageSignal = new SuppliedDoubleSignal("main motor voltage", motorsVoltage);
```

Comment:
main motor?

Reply:
(1) there isn't a main motor, mechanics suggested a reserved & main motor since they'll probably remove one of them but anyway @YHerm hates calling it "main motor" (see [disscustion](https://github.com/GreenBlitz/2024-Offseason-robot/pull/8#discussion_r1784235976))

Reply:
💀 you wrote main motor in this line

Reply:
I pushed a fixed around 0.5mim after your wrote that comment